### ISSUE TITLE
PCA Features Outputs different Number of Features on Train and Test Set

### ISSUE THREAD

**Issue by BlackArbsCEO** [hudson-and-thames/mlfinlab#248]
Because the number of features cannot be specified ahead of time, the number of features changes between train and test sets. This wouldn't be a problem except scikit-learn classifiers require the same number of features between train and test sets.

Here is the error that's generated:

```
ValueError: Number of features of the model must match the input. Model n_features is 22 and input n_features is 16.
```

My versions
```
numpy 1.18.1
pandas 1.0.3
scipy 1.4.1
sklearn 0.22.1
mlfinlab 0.7.3
```

**Comment by proskurin** [comment]
Hello @BlackArbsCEO, could you please provide the whole code snippet?

**Comment by BlackArbsCEO** [comment]
```
def get_pca_features(in_X):
    pca_features = pd.DataFrame(
        ml.feature_importance.orthogonal.get_orthogonal_features(in_X), index=in_X.index
    )
    pca_features.columns = [f"pca_{i}" for i in range(len(pca_features.columns))]
    return pca_features


pca_train= get_pca_features(X_train)
pca_test= get_pca_features(X_test)

print(pca_train.shape) 
print(pca_test.shape)

# (1874, 22)
# (799, 15)
```

**Comment by PanPip** [comment]
Now `ml.feature_importance.orthogonal.get_orthogonal_features` function allows input of `num_features` parameter that allows specifying the number of features to get from PCA (this overrides the `variance_thresh` input variable).

So, to get the same number of features in the test set one can use:

<pre>
def get_pca_features(in_X, num_features=None):
    pca_features = pd.DataFrame(
        ml.feature_importance.orthogonal.get_orthogonal_features(in_X, num_features=num_features), index=in_X.index
    )
    pca_features.columns = [f"pca_{i}" for i in range(len(pca_features.columns))]
    return pca_features


pca_train= get_pca_features(X_train)
pca_test= get_pca_features(X_test,num_features=8)

</pre>

This issue is fixed in PR #402.
